util/set.go

```diff
@@ -40,3 +40,37 @@ func (s Set[T]) Contains(elem T) (exists bool) {
 	_, exists = s[elem]
 	return
 }
+
+// Union constructs a new set, containing all elements from the given sets. nil
+// is never returned
+func Union[T comparable](sets ...Set[T]) Set[T] {
+	union := make(Set[T])
+	for _, set := range sets {
+		for elem := range set {
+			union.Add(elem)
+		}
+	}
+	return union
+}
+
+// Intersection constructs a new set, containing all elements that appear in all
+// given sets. nil is never returned.
```

Comment:
This is pedantic, but could you add a comment saying that the intersection of zero sets is an empty set?  If there are zero of "all given sets", then pedantically speaking, you could argue that every value of type T is in those zero "all given sets", but it's not in the resulting set.

Reply:
I'd be happy to.